packages/kit/src/runtime/client/client.js

```diff
@@ -1531,11 +1531,22 @@ export function create_client({ target, base }) {
 					// with history.go, which means we end up back here, hence this check
 					if (event.state[INDEX_KEY] === current_history_index) return;
 
+					const scroll = scroll_positions[event.state[INDEX_KEY]];
+
+					// if the only change is the hash, we don't need to do anything...
+					if (current.url.href.split('#')[0] === location.href.split('#')[0]) {
```

Comment:
Do we need to worry about the case where the old or the new (or both!) URLs don't have a `#` portion? Are we treating a link from `/foo` to `/foo` as a no-op? (I remember this coming up before, but don't remember what the decision was.) What happens in native browser navigation when there's a link from `/foo#bar` to `/foo`?

Reply:
Native navigation: 
- `/foo#bar` to `/foo` (or the other way around) only resets the scroll position
- `/foo` to `/foo` doesn't create a new history entry, so that case doesn't occur
--> maybe the solution is to use `replaceState` in `update` in case the URL is the same? Then we don't need to worry about these cases

Reply:
yep, we're in the clear — this check will work the same for `/foo#x` <-> `/foo` as for `/foo#x` <-> `/foo#y`, and we already treat `/foo` <-> `/foo` as a `replaceState` inside the `click` handler:

https://github.com/sveltejs/kit/blob/1a0f82a9c953ba3370628811d39f1e73ea3d08f5/packages/kit/src/runtime/client/client.js#L1465